cmd/rook/ceph/osd.go

```diff
@@ -418,35 +417,50 @@ func readCephSecret(path string) error {
 	return nil
 }
 
-func destroyOSD(context *clusterd.Context, clusterInfo *client.ClusterInfo, osdID int) (*oposd.OSDInfo, error) {
-	osdInfo, err := osddaemon.GetOSDInfoById(context, clusterInfo, osdID)
+// destroyOSD fetches the OSD to be replaced based on the ID and then destroys that OSD and zaps the backing device
+func destroyOSD(context *clusterd.Context, clusterInfo *client.ClusterInfo, id int) (*osd.OSDReplaceInfo, error) {
```

Comment:
How about moving this method into the package `pkg/daemon/ceph/osd`?

Reply:
Could it be part of the DiskSanitizer?

Reply:
This could go in the existing `pkg/daemon/ceph/osd/remove.go`  or a new `pkg/daemon/ceph/osd/destroy.go` 

Problem with `diskSanitizer` is that it does not handle any error. So modifications will be needed in order to user existing methods.

Reply:
DiskSanitizer is mostly about cleaning the disks and controlling how the disks can be cleaned using the `SanitizeDisksSpec`.  

We need to destroy OSD and run `ceph-volume zap` to clean the disk. We don't need any fine grained control over how the disk can be cleaned by using the `SanitizeDisksSpec`. 

So I feel `DiskSanitizer` might not be the best place to have this method.

Reply:
Ok, then I'd vote for `pkg/daemon/ceph/osd/remove.go`.

Reply:
moved the `DestroyOSD` method to `pkg/daemon/ceph/osd/remove.go`